WordPress/gutenberg
# Patterns: Set width of synced pattern editor block wrapper to max width of widest child

## What?
Checks the align setting of child blocks of a synced pattern and sets the editor block wrapper to the same so that the width is respected in the editor. Also adds layout support of type `constrained` to block wrapper if wide or full align are found to ensure that alignments are correctly applied.

## Why?
Because synced blocks have a wrapper in the editor, but not the front end, the full and max-width settings are not respected.

## How?
Find the widest align option of the child blocks and apply this same align class to the parent wrapper.

## Testing Instructions

- Add a Cover block and set the alignment to full-width
- Covert to a Synced pattern and make sure the full width still applied in the editor and frontend
- Try different blocks with different alignments and make sure the widths are respected in the editor and frontend
- Try changing the width/align of the cover block after  it has been converted to a synced pattern and make sure correct alignment is applied
- Add a group block with width set to max width and add a cover block to it that has width set to full. Convert to synced pattern and make sure cover block is still constrained by width of parent group block

## Screenshots or screencast 
Before:

https://github.com/WordPress/gutenberg/assets/3629020/48e659cc-99f4-45f9-b262-627d3daa78ab

After:

https://github.com/WordPress/gutenberg/assets/3629020/2787e343-421b-4622-9186-93124cd3b292
